File: Front-end/BusinessLayer/Contracts/BaseResponse.js
export class BaseResponse
{
    constructor(data="")
    {
        this.data = data;
        this.error={errorMessage:"",hasError:false};
    }

    addError(message)
    {
        this.error.hasError=true;
        if(message)
        {
            this.error.errorMessage = message;
        }
    }
}

File: Back-end/BusinessLayer/Contracts/BaseResponse.cs
namespace BusinessLayer.Contracts;

public class BaseResponse
{
    public BaseResponse()
    {
        Error = new ErrorResponse();
    }

    public ErrorResponse Error { get; }

    public void AddError(string error)
    {
        Error.HasError = true;
        Error.ErrorMessage = error;
    }

    public bool HasError()
    {
        return Error.HasError;
    }
}
